source/PlanetPanel.cpp

```diff
@@ -70,6 +70,16 @@ PlanetPanel::PlanetPanel(PlayerInfo &player, function<void()> callback)
 
 void PlanetPanel::Step()
 {
+	// If the player is teleporting, simulate a TakeOff followed
+	// by a forced landing. No transition will be noticeable.
+	if(player.RelocationStatus() == PlayerInfo::RelocateStatus::TELEPORTING)
+	{
+		player.SetRelocationStatus(PlayerInfo::RelocateStatus::TELEPORTED);
+		UI *ui = GetUI();
+		while(ui->IsTop(this))
+			ui->Pop(ui->Top().get());
+		TakeOff();
```

Comment:
Do we still need to go through TakeOff()? That function is very small, so it might be  more clear to just not re-use take-off but to implement the small part of TakeOff() that still is used here.

Reply:
done with next commit